extensions/wikia/AdEngine/js/provider/gpt/adElement.js

```diff
@@ -21,9 +20,9 @@ define('ext.wikia.adEngine.provider.gpt.adElement', [
 			this.node.id = this.id;
 		}
 		if (slotTargeting.size) {
-			this.sizes = adSizeConverter.convert(slotTargeting.size);
-			this.sizes = adSizeFilter.filter(slotName, this.sizes);
+			this.sizes = adSizeFilter.filter(slotName, slotTargeting.size);
```

Comment:
IMO it's cleaner to filter and convert it separately. Right now your `adSizeFilter` knows and do too much

Reply:
I would generally agree, but in this case it's' much more convenient to convert it inside `filterSizes` method. For some of the filtering usecases it's easier to operate on string, whereas for other it's more convenient to iterate over array, that's why conversion of the format is done in such way. I'll try to split those operations for bigger readability though